### PR TITLE
feat(earn): Add UI skeleton of earn collect

### PR DESCRIPTION
**Pull request by jh2oman**
### Description

[Figma](https://www.figma.com/design/E1rC3MG74qEg5V4tvbeUnU/Stablecoin-Enablement?node-id=2472-14441&m=dev)

TODO in followup PR: 
* unit tests
* hook into Earn flow
* fetch real data for rewards, apy
* prepare transactions, estimate gas, execute transactions
* loading, disabled, error states.

Screenshot_20240513_235415

### Test plan

Unit tests in followup PR

### Related issues

https://linear.app/valora/issue/ACT-1180/wallet-earn-collect-review-screen

### CODE REVIEW

File: locales/base/translation.json [hunk 1]
```diff
@@ -2369,6 +2369,15 @@
       "footer": "By depositing crypto into an Aave pool, you accept the risks associated with using Aave. <0>Terms & Conditions</0>",
       "primaryCta": "Complete",
       "secondaryCta": "Cancel"
+    },
+    "collect": {
+      "title": "Congratulations! Review & collect your earnings",
+      "total": "Total",
+      "plus": "Plus",
```
Comment (satish-ravi) [inline locales/base/translation.json:2376]: these are upper cases in figma

File: src/earn/EarnCollectScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+        <Text style={styles.title}>{t('earnFlow.collect.title')}</Text>
+        <View style={styles.collectInfoContainer}>
+          <CollectItem
+            title={t('earnFlow.collect.total')}
+            tokenInfo={tokenInfo}
+            rewardAmount={poolTokenInfo.balance}
+          />
+          {rewardsInfo.map((info, index) => (
+            <CollectItem
+              title={t('earnFlow.collect.total')}
```
Comment (satish-ravi) [inline src/earn/EarnCollectScreen.tsx:79]: total -> plus

File: src/earn/EarnCollectScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+    ...typeScale.labelSemiBoldLarge,
+    color: Colors.black,
+  },
+  fiatText: {
+    ...typeScale.bodySmall,
+    color: Colors.gray4,
+  },
+  collectItemTitle: {
+    ...typeScale.labelSemiBoldXSmall,
+    marginBottom: 4,
```
Comment (satish-ravi) [inline src/earn/EarnCollectScreen.tsx:216]: use Spacing.Tiny4 here and below. Also, some of the padding looks off from the figma

File: src/earn/EarnCollectScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+      <Text style={styles.collectItemTitle}>{title}</Text>
+      <View style={styles.row}>
+        <View style={styles.iconContainer}>
+          <TokenIcon token={tokenInfo} size={IconSize.LARGE} />
+        </View>
+        <View>
+          <TokenDisplay
+            style={styles.cryptoText}
+            tokenId={tokenInfo.tokenId}
+            amount={rewardAmount || tokenInfo.balance}
```
Comment (satish-ravi) [inline src/earn/EarnCollectScreen.tsx:121]: Do we need this fallback? What's a case where it can be falsy?

Comment (jh2oman) [reply src/earn/EarnCollectScreen.tsx:121]: from a previous iteration

**Review by satish-ravi — APPROVED** [review]

**Pull request merged**